### PR TITLE
docs: improve HAR example

### PR DESCRIPTION
**Pull request by debs-obrien**
improve the HAR example and explanation:
- other langs not added yet as will depend on feedback from js examples first

### CODE REVIEW

File: docs/src/network.md [hunk 1]
```diff
@@ -791,6 +791,28 @@ var context = await browser.NewContextAsync(new() {
...
+### Recording HAR from a test
+
+To record a HAR file we use the `routeFromHAR` method. This method takes in the path to the HAR file and an optional object of options.
+
+The options object can contain the URL so that only requests with the URL matching the specified glob pattern will be served from the HAR File. If not specified, all requests will be served from the HAR file.
+
+The `update` option updates the given HAR file with the actual network information instead of serving from the file. In order to record the HAR file, you need to set `update` to true.
+
+```js
+test('Record a HAR', async ({
```
Comment (dgozman) [inline docs/src/network.md:803]: We don't usually split test header into multiple lines in other examples. Let's be consistent.

File: docs/src/network.md [hunk 2]
```diff
@@ -826,6 +848,25 @@ page.route_from_har("example.har")
 await context.RouteFromHARAsync("example.har");
 ```
 
+### Mocking with HAR
+
+You can modify the response from the HAR file and then run your tests against the modified response.
```
Comment (dgozman) [inline docs/src/network.md:853]: Not sure how the snippet below is different from the previous sections, nor how does it highlight "you can modify the response" part?

**Review by dgozman — COMMENTED** [review]
Should we move HAR record/replay into mocking APIs doc?

File: docs/src/network.md (hunk 2 above)
Comment (debs-obrien) [reply docs/src/network.md:853]: yeh I need to add that they need to modify the .txt file to modify the response, will fix

**Comment by debs-obrien** [comment]
closing as this is now dealt with in this pr: https://github.com/microsoft/playwright/pull/23759

**Pull request closed without merging**